Translated Content. (#2)
1950Labs/docs-next: src/guide/installation.md

# Installation
# Instalación

> Vue.js is built by design to be incrementally adoptable. This means that it can be integrated into a project multiple ways depending on the requirements.

Reviewer: I think the translation for this piece is missing 🤔

Author: Updated. It was added after I translated it. [4b0ca9](https://github.com/vuejs/docs-next/commit/4b0ca9a03db951fe1bed6055cafbfda7d6be6001#diff-923f699fa88653104589aefed42ad89c3b1df393cdd375c0e0805f4e305ab93e)

Reviewer: Awesome, thanks!

## Notas de Lanzamiento

There are three primary ways of adding Vue.js to a project:
Última Versión: ![npm](https://img.shields.io/npm/v/vue/next.svg)

1. Import it as a [CDN package](#cdn) on the page
2. Install it using [npm](#npm)

> 3. Use the official [CLI](#cli) to scaffold a project, which provides batteries-included build setups for a modern frontend workflow (e.g., hot-reload, lint-on-save, and much more)

Reviewer: I think the translation for this piece is missing 🤔

Author: Updated. It was added after I translated it. [4b0ca9](https://github.com/vuejs/docs-next/commit/4b0ca9a03db951fe1bed6055cafbfda7d6be6001#diff-923f699fa88653104589aefed42ad89c3b1df393cdd375c0e0805f4e305ab93e)

Reviewer: Awesome, thanks!

## CLI

> Vue provee una [interfaz de línea de comandos (CLI) oficial](https://github.com/vuejs/vue-cli) para realizar la creación rápida de la estructura base para las Single Page Applications. Esta provee _setups_ con baterías incluidas para un flujo de trabajo de frontend moderno. Solo toma unos minutos configurar y ejecutarlo con _hot-reload_, _lint-on-save_, y compilaciones listas para producción. Véase [the Vue CLI docs](https://cli.vuejs.org) para más detalles.

Reviewer: What do you think about this?
```suggestion
Vue provee una [interfaz de línea de comandos (CLI) oficial](https://github.com/vuejs/vue-cli) para realizar la creación rápida de la estructura base para las Single Page Applications. Proporciona configuraciones de _build_ prontas para un proceso de trabajo de _front-end_ moderno. Solo toma unos minutos configurar y ejecutarlo con _hot-reload_, _lint-on-save_, y compilaciones listas para producción. Véase [the Vue CLI docs](https://cli.vuejs.org) para más detalles.
```

Author: Maybe we could write something like this. @Tintef @sxcamacho @migsalazar What do you guys think?

```suggestion
Vue provee una [interfaz de línea de comandos (CLI) oficial](https://github.com/vuejs/vue-cli) para realizar la creación rápida de la estructura base para las Single Page Applications. Proporciona configuraciones de _build_ con baterías incluidas para un proceso de trabajo de _front-end_ moderno. Solo toma unos minutos configurar y ejecutarlo con _hot-reload_, _lint-on-save_, y compilaciones listas para producción. Véase [the Vue CLI docs](https://cli.vuejs.org) para más detalles.
```

Reviewer 2: The current docs are using `prontas`, so maybe that's the way.

But, IMHO, `prontas` isn't clear for me (topic-related: spanish in different countries). If I may make a suggestion, maybe something like this could work:

```suggestion
Vue provee una [interfaz de línea de comandos (CLI) oficial](https://github.com/vuejs/vue-cli) para realizar la creación rápida de la estructura base para las Single Page Applications. Proporciona configuraciones de _build_ listas para su uso en un proceso de trabajo de _front-end_ moderno. Solo toma unos minutos configurar y ejecutarlo con _hot-reload_, _lint-on-save_, y compilaciones listas para producción. Véase [the Vue CLI docs](https://cli.vuejs.org) para más detalles.
```

Reviewer: I like the one @migsalazar proposed the most. I'm not a big fan of using `baterías`, it sounds weird in Spanish IMO

Author: Yes, @migsalazar got a better one!

::: tip

> La interfaz de línea de comandos (CLI) asume que usted posee conocimiento previo de Node.js y las herramientas de compilación asociadas. Si usted es nuevo en Vue o en el manejo de las herramientas de compilación de front-end, le recomendamos leer <a href="./">la guía</a> sin ninguna herramienta de compilación antes de usar la interfaz de línea de comandos.

Reviewer: We should use italics for `front-end`
```suggestion
La interfaz de línea de comandos (CLI) asume que usted posee conocimiento previo de Node.js y las herramientas de compilación asociadas. Si usted es nuevo en Vue o en el manejo de las herramientas de compilación de _front-end_, le recomendamos leer <a href="./">la guía</a> sin ninguna herramienta de compilación antes de usar la interfaz de línea de comandos.
```

:::

> Para Vue 3, usted debería utilizar Vue CLI v4.5 disponible en `npm` bajo el nombre de `@vue/cli`. Para actualizar, ustd necesita reinstalar la última versión de `@vue/cli` globalmente:

Reviewer: Typo
```suggestion
Para Vue 3, usted debería utilizar Vue CLI v4.5 disponible en `npm` bajo el nombre de `@vue/cli`. Para actualizar, usted necesita reinstalar la última versión de `@vue/cli` globalmente:
```

# O
npm install -g @vue/cli
```

Luego, en los proyectos de Vue, ejecute

```bash
vue upgrade --next
```

## Vite

[Vite](https://github.com/vitejs/vite) es una herramienta de desarrollo web que le permite el rápido despliegue del código debido a su enfoque nativo para importación de Módulos ES.

Los proyectos de Vue pueden ser rápidamente configurados con Vite ejecutando los siguientes comandos en su terminal.

Con NPM:

```bash
$ npm init vite-app <nombre-de-proyecto>
$ cd <nombre-de-proyecto>
$ npm install
$ npm run dev
```

O con Yarn:

```bash
$ yarn create vite-app <nombre-de-proyecto>
$ cd <nombre-de-proyecto>
$ yarn
$ yarn dev
```

## Explicación de las Distintas Compilaciones.

En el [directorio `dist/` del paquete NPM](https://cdn.jsdelivr.net/npm/vue@3.0.0-rc.1/dist/) encontrará varias compilaciones de Vue.js. Aquí está una visión general de cuál archivo del directorio `dist` debería ser utilizado dependiendo del caso de uso.

### Desde un CDN o sin un Empaquetador

#### `vue(.runtime).global(.prod).js`:

- Para uso directo mediante `<script src="...">` en el navegador, expone la variable Vue globalmente.
- Compilación de plantillas dentro del navegador:

>   - `vue.global.js` es la compilación completa que incluye ambos, el compilador y el entorno de ejecución, así que soporta compilar plantillas sobre la marcha.

Reviewer: What do you think about this?
```suggestion
  - `vue.global.js` es la compilación completa que incluye tanto el compilador como el entorno de ejecución, así que soporta compilar plantillas sobre la marcha.
```

Author: I think it fits better in this context.

  - `vue.runtime.global.js` contiene solo el entorno de ejecución y requiere que las plantillas sean pre-compiladas durante la etapa de compilación.

> - Todos los paquetes internos del núcleo de Vue en líneas, p. ej. un único archivo sin dependencias de otros. Esto significa que debe importar todo solo desde este archivo para garantizar que está obteniendo la misma instancia de código.

Reviewer: I'm not a big fan of abbreviating `por ejemplo` as `p. ej.`, it's not really normal in Spanish. 
```suggestion
- Todos los paquetes internos del núcleo de Vue, por ejemplo un único archivo sin dependencias de otros. Esto significa que usted debe importar todo solamente desde este archivo para garantizar que está obteniendo la misma instancia de código.
```

Author: Yes, it's better this way. We want to be as clear as possible :smile:.

Author: Corrected `solomente` to `solamente`.

- Contiene las ramas prod/dev codificadas, y la compilación de prod está pre-minificada. Utilice los archivos `*.prod.js` para producción.

:::tip Nota
Las compilaciones globales no son compilaciones [UMD](https://github.com/umdjs/umd). Están compiladas como [IIFEs](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) y solo están destinadas para uso directo mediante `<script src="...">`.
:::

#### vue(.runtime).esm-browser(.prod).js:

- Para uso mediante la importación de módulos ES nativos (en el navegador mediante `<script type="module">`).
- Comparte la misma compilación del entorno de ejecución, declaración de dependencias en línea y el comportamiento codificado prod/dev con la compilación global.

### Con un Empaquetador

#### vue(.runtime).esm-bundler.js:

- Para uso con empaquetadores como `webpack`, `rollup` y `parcel`.

> - Deja las ramas prod/dev con `process.env.NODE_ENV guards` (debe ser reemplazado por el empaquetador)

Reviewer: I think we can translate `guards` as `guardas`. What do you think? @sxcamacho ?

- No entrega las compilaciones minificadas (para ser minificado junto con el resto del código después de empaquetar)

> - Importa las dependencias (p. ej. `@vue/runtime-core`, `@vue/runtime-compiler`)

Reviewer: Same here about the abbreviation
```suggestion
- Importa las dependencias (por ejemplo `@vue/runtime-core`, `@vue/runtime-compiler`)
```

>   - Las dependencias importadas son también compilaciones _esm-bundler_ y es por esto que importarán sus dependencias (e.g. @vue/runtime-core imports @vue/reactivity)

Reviewer: Missing translation:
```suggestion
  - Las dependencias importadas son también compilaciones _esm-bundler_ y es por esto que importarán sus dependencias (por ejemplo @vue/runtime-core imports @vue/reactivity)
```

  - Esto significa que usted **puede** instalar/importar esas dependencias sin terminar con diferentes instancias de ellas, pero debe estar seguro de que todas ellas se resuelven a la misma versión.
- Compilación de plantillas dentro del navegador:

>   - `vue.runtime.esm-bundler.js` **(por defecto)** es solo entorno de ejecución, y requiere que todas las plantillas sean pre-compiladas. Esta es la entrada por defecto para los empaquetadores (mediante el campo módulo en `package.json`) debido a que cuando se utiliza un empaquetador las plantillas son típicamente pre-compiladas (p. ej. en los archivos `*.vue`).

Reviewer: Same here about the abbreviation

```suggestion
  - `vue.runtime.esm-bundler.js` **(por defecto)** es solo entorno de ejecución, y requiere que todas las plantillas sean pre-compiladas. Esta es la entrada por defecto para los empaquetadores (mediante el campo módulo en `package.json`) debido a que cuando se utiliza un empaquetador las plantillas son típicamente pre-compiladas (por ejemplo en los archivos `*.vue`).
```

>   - `vue.esm-bundler.js`: incluye el compilador del entorno de ejecución. Use esto si está empleando un empaquetador pero quiere la compilación de plantillas en tiempo de ejecución (p. ej. plantillas dentro del DOM o plantillas mediante _strings_ en línea con JavaScript). Necesitará configurar su empaquetador para establecer el alias vue a este archivo.

Reviewer: Same here about the abbreviation

```suggestion
  - `vue.esm-bundler.js`: incluye el compilador del entorno de ejecución. Use esto si está empleando un empaquetador pero quiere la compilación de plantillas en tiempo de ejecución (por ejemplo plantillas dentro del DOM o plantillas mediante _strings_ en línea con JavaScript). Necesitará configurar su empaquetador para establecer el alias vue a este archivo.
```

### Para el Renderizado en el lado del Servidor

#### `vue.cjs(.prod).js`:

- Para uso en renderizado en el lado del servidor con Node.js mediante `require()`.
- Si usted empaqueta su aplicación con webpack utilizando `target: 'node'` y externaliza apropiadamente `vue`, esta será la compilación que será cargada.
- Los archivos dev/prod son pre-compilados, pero el archivo apropiado es requerido automáticamente basado en `process.env.NODE_ENV`.

## Entorno en tiempo de Ejecución + Compilador vs. Solo Entorno en tiempo de Ejecución

> Si usted necesita compilar plantillas en el cliente (p. ej. pasando un parámetro a la opción de plantilla, o montando en un elemento utilizando su HTML dentro del DOM como la plantilla), necesitará el compilador y por ello la compilación completa:

Reviewer: Same here about the abbreviation

```suggestion
Si usted necesita compilar plantillas en el cliente (por ejemplo pasando un parámetro a la opción de plantilla, o montando en un elemento utilizando su HTML dentro del DOM como la plantilla), necesitará el compilador y por ello la compilación completa:
```

```js
// esto requiere el compilador
Vue.createApp({

>   template: '<div>{{ hola }}</div>'

Reviewer: As a thumb rule from the previous docs, we don't translate code (**we do translate comments**)

Author: Got it!

>   template: '<div>{{ hola }}</div>'

Author: ```suggestion
  template: '<div>{{ hi }}</div>'
```

>   template: '<div>{{ hola }}</div>'

Author: ````suggestion
  template: '<div>{{ hi }}</div>'
```

})

// esto no
Vue.createApp({
  render() {

>     return Vue.h('div', {}, this.hola)

Reviewer: As a thumb rule from the previous docs, we don't translate code (**we do translate comments**)

Author: ```suggestion
    return Vue.h('div', {}, this.hi)
```
